internal/dag/cache.go

```diff
@@ -288,7 +310,7 @@ func (kc *KubernetesCache) remove(obj interface{}) bool {
 // serviceTriggersRebuild returns true if this service is referenced
 // by an Ingress or HTTPProxy in this cache.
 func (kc *KubernetesCache) serviceTriggersRebuild(service *v1.Service) bool {
-	for _, ingress := range kc.ingresses {
+	for _, ingress := range kc.ingressesv1beta1 {
```

Comment:
I think we're going to need to check both `kc.ingresses` and `kc.ingressesv1beta1`, since either one of them could be populated based on whether `v1` exists in the cluster or not.